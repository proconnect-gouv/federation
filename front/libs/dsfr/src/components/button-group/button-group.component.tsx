import classnames from 'classnames';
import type { PropsWithChildren } from 'react';
import React, { Children } from 'react';

import { Sizes } from '../../enums';

interface ButtonGroupComponentProps extends PropsWithChildren {
  size?: Sizes;
  inline?: boolean;
  equisized?: boolean;
}

export const ButtonGroupComponent = React.memo(
  ({
    children,
    equisized = false,
    inline = true,
    size = Sizes.MEDIUM,
  }: ButtonGroupComponentProps) => (
    <ul
      className={classnames('fr-btns-group', {
        [`fr-btns-group--inline-${size}`]: inline,
        [`fr-btns-group--${size}`]: !inline,
        // css class from DSFR
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-btns-group--equisized': equisized,
      })}>
      {Children.map(children, (child) => (
        <li>{child}</li>
      ))}
    </ul>
  ),
);

ButtonGroupComponent.displayName = 'ButtonGroupComponent';
