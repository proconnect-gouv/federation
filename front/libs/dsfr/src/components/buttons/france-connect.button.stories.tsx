import { ComponentMeta } from '@storybook/react';

import { FranceConnectButton, FranceConnectButtonProps } from './france-connect.button';

export default {
  component: FranceConnectButton,
  title: 'DSFR/components/buttons/FranceConnectButton',
} as ComponentMeta<typeof FranceConnectButton>;

export const Default = (args: FranceConnectButtonProps) => <FranceConnectButton {...args} />;
Default.args = {} as FranceConnectButtonProps;
