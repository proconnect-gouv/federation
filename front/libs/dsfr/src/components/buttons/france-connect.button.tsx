import classnames from 'classnames';
import React from 'react';

import { ButtonTypes } from '../../enums';

interface FranceConnectButtonProps {
  showIcon?: boolean;
  className?: string;
  showHelp?: boolean;
  buttonType?: ButtonTypes;
}

export const FranceConnectButton = React.memo(
  ({
    buttonType = ButtonTypes.BUTTON,
    className,
    showHelp = false,
    showIcon = false,
  }: FranceConnectButtonProps) => {
    const target = showIcon ? '_blank' : undefined;
    const rel = showIcon ? 'noreferrer' : undefined;
    return (
      <div className={classnames('fr-connect-group', className)}>
        <button className="fr-connect" type={buttonType}>
          <span className="fr-connect__login">S’identifier avec</span>
          <span className="fr-connect__brand">FranceConnect</span>
        </button>
        {showHelp && (
          <p>
            <a
              href="https://franceconnect.gouv.fr/"
              rel={rel}
              target={target}
              title="Qu’est ce que FranceConnect ? - nouvelle fenêtre">
              Qu’est ce que FranceConnect ?
            </a>
          </p>
        )}
      </div>
    );
  },
);

FranceConnectButton.displayName = 'FranceConnectButton';
