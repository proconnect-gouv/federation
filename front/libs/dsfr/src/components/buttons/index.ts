export * from './link';
export * from './login-connect';
export * from './simple';
