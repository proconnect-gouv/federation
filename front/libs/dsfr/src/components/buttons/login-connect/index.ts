export * from './login-connect.component';
