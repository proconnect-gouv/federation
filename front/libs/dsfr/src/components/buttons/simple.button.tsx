import classnames from 'classnames';
import React, { ButtonHTMLAttributes } from 'react';

import { IconPlacement, Priorities, Sizes } from '../../enums';

interface SimpleButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  // @SEE https://gouvfr.atlassian.net/wiki/spaces/DB/pages/217284660/Boutons+-+Buttons
  label: string;
  size?: Sizes;
  priority?: Priorities;
  noOutline?: boolean;
  icon?: string;
  iconPlacement?: IconPlacement;
}

export const SimpleButton: React.FC<SimpleButtonProps> = React.memo(
  ({
    disabled,
    icon,
    iconPlacement,
    label,
    noOutline,
    onClick,
    priority,
    size,
    title,
    type: nativeButtonType,
  }: SimpleButtonProps) => (
    <button
      className={classnames(`fr-btn fr-btn--${size}`, {
        // DSFR classname
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-btn--secondary': priority === Priorities.SECONDARY,
        // DSFR classname
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-btn--tertiary': priority === Priorities.TERTIARY && !noOutline,
        // DSFR classname
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-btn--tertiary-no-outline': priority === Priorities.TERTIARY && noOutline,
        [`fr-btn--icon-${iconPlacement}`]: !!icon,
        [`fr-fi-${icon}`]: !!icon,
      })}
      disabled={disabled}
      title={title}
      type={nativeButtonType}
      onClick={onClick}>
      {label}
    </button>
  ),
);

SimpleButton.defaultProps = {
  icon: undefined,
  iconPlacement: IconPlacement.RIGHT,
  noOutline: false,
  priority: Priorities.PRIMARY,
  size: Sizes.MEDIUM,
};

SimpleButton.displayName = 'SimpleButton';
