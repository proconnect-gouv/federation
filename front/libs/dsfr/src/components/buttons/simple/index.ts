/* istanbul ignore file */

// declarative file
export * from './simple.button';
