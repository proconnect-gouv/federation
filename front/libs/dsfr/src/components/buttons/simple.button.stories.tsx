import { ComponentMeta } from '@storybook/react';

import { Priorities, Sizes } from '../../enums';
import { SimpleButton, SimpleButtonProps } from './simple.button';

export default {
  component: SimpleButton,
  title: 'DSFR/components/buttons/SimpleButton',
} as ComponentMeta<typeof SimpleButton>;

export const Default = (args: SimpleButtonProps) => <SimpleButton {...args} />;

Default.args = {
  label: 'Simple Button',
  noOutline: false,
  priority: Priorities.PRIMARY,
  size: Sizes.MEDIUM,
} as SimpleButtonProps;
