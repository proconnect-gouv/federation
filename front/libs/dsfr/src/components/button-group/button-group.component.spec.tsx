import { render } from '@testing-library/react';

import { SimpleButton } from '../buttons';
import { ButtonGroupComponent } from './button-group.component';

jest.mock('../buttons/simple/simple.button');

describe('ButtonGroupComponent', () => {
  it('should match snapshot with default props', () => {
    // when
    const { container } = render(
      <ButtonGroupComponent>
        <SimpleButton label="précédent" />
        <SimpleButton label="suivant" />
      </ButtonGroupComponent>,
    );

    // then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('fr-btns-group');
    expect(container.firstChild).toHaveClass('fr-btns-group--inline-md');
    expect(container.firstChild).not.toHaveClass('fr-btns-group--equisized');
    expect(SimpleButton).toHaveBeenCalledTimes(2);
  });

  it('should match snapshot with all props', () => {
    // when
    const { container } = render(
      <ButtonGroupComponent equisized inline={false}>
        <SimpleButton label="précédent" />
        <SimpleButton label="suivant" />
      </ButtonGroupComponent>,
    );

    // then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('fr-btns-group');
    expect(container.firstChild).toHaveClass('fr-btns-group--equisized');
    expect(container.firstChild).toHaveClass('fr-btns-group--md');
    expect(SimpleButton).toHaveBeenCalledTimes(2);
  });
});
