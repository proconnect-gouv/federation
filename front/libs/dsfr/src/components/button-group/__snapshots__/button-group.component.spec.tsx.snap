// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroupComponent should match snapshot with all props 1`] = `
<div>
  <ul
    class="fr-btns-group fr-btns-group--md fr-btns-group--equisized"
  >
    <li>
      <div
        data-mockid="SimpleButton"
      />
    </li>
    <li>
      <div
        data-mockid="SimpleButton"
      />
    </li>
  </ul>
</div>
`;

exports[`ButtonGroupComponent should match snapshot with default props 1`] = `
<div>
  <ul
    class="fr-btns-group fr-btns-group--inline-md"
  >
    <li>
      <div
        data-mockid="SimpleButton"
      />
    </li>
    <li>
      <div
        data-mockid="SimpleButton"
      />
    </li>
  </ul>
</div>
`;
